default.nix

```diff
@@ -1,289 +1,28 @@
-          sha256 = "12b4xi6nlnhpwz8apn4mk880mkhcv1sfvf4j3z1h5dgkadi2zgbi";
-        });
-        case-insensitive = overrideCabal super.case-insensitive (drv: {
-          version = "1.2.0.4";
-          sha256 = "07nm40r9yw2p9qsfp3pjbsmyn4dabrxw34p48171zmccdd5hv0v3";
-        });
-        dlist = overrideCabal super.dlist (drv: {
-          version = "0.7.1.1";
-          sha256 = "1zayvxvkan2s2ixajdr3f5rn1gzhprzv6cww4cbpwjhzw0l7zc08";
-        });
-        extensible-exceptions = overrideCabal super.extensible-exceptions (drv: {
-          version = "0.1.1.3";
-          sha256 = "1i8rjfczsx1wjfaq423a7cp7qrnxh053865z7bg6hwhk2pxsrxkm";
-        });
-        hashable = overrideCabal super.hashable (drv: {
-          version = "1.2.3.2";
-          sha256 = "0h9295pv2sgbaqlwpwbx2bap6nngm0jcdhkqham1wpjwyxqgqrlc";
-        });
-        mtl = overrideCabal super.mtl (drv: {
-          version = "2.2.1";
-          sha256 = "1icdbj2rshzn0m1zz5wa7v3xvkf6qw811p4s7jgqwvx1ydwrvrfa";
-        });
-        old-locale = overrideCabal super.old-locale (drv: {
-          version = "1.0.0.7";
-          sha256 = "0l3viphiszvz5wqzg7a45zp40grwlab941q5ay29iyw8p3v8pbyv";
-        });
-        old-time = overrideCabal super.old-time (drv: {
-          version = "1.1.0.3";
-          sha256 = "1h9b26s3kfh2k0ih4383w90ibji6n0iwamxp6rfp2lbq1y5ibjqw";
-        });
-        parallel = overrideCabal super.parallel (drv: {
-          version = "3.2.0.6";
-          sha256 = "0hp6vf4zxsw6vz6lj505xihmnfhgjp39c9q7nyzlgcmps3xx6a5r";
-        });
-        primitive = overrideCabal super.primitive (drv: {
-          version = "0.5.4.0";
-          sha256 = "05gdgj383xdrdkhxh26imlvs8ji0z28ny38ms9snpvv5i8l2lg10";
-          revision = "1";
-          editedCabalFile = "df0a129c168c61a06a02123898de081b1d0b254cce6d7ab24b8f43ec37baef9e";
-          });
-        scientific = overrideCabal super.scientific (drv: {
-          version = "0.3.3.3";
-          sha256 = "1hngkmd1kggc84sz4mddc0yj2vyzc87dz5dkkywjgxczys51mhqn";
-          jailbreak = true;
-        });
-        stm = overrideCabal super.stm (drv: {
-          version = "2.4.4";
-          sha256 = "0gc8zvdijp3rwmidkpxv76b4i0dc8dw6nbd92rxl4vxl0655iysx";
-        });
-        syb = overrideCabal super.syb (drv: {
-          version = "0.5.1";
-          sha256 = "0iiqz5mamk1nsij99rypms7dhx5flm2n02k1x6miqgnhg075zc41";
-        });
-        unordered-containers = overrideCabal super.unordered-containers (drv: {
-          version = "0.2.5.1";
-          sha256 = "06l1xv7vhpxly75saxdrbc6p2zlgz1az278arfkz4rgawfnphn3f";
-        });
-        vector = overrideCabal super.vector (drv: {
-          version = "0.10.12.2";
-          sha256 = "01hc71k1z9m0g0dv4zsvq5d2dvbgyc5p01hryw5c53792yi2fm25";
-          jailbreak = true;
-        });
+let nixpkgs = pkgs ({
+      config = { allowUnfree = true; } //
+      (if config == null then {} else config); } //
+      (if system == null then {} else { inherit system; }));
```

Comment:
I like the change to put this stuff on fewer lines, but I find this new indentation confusing: the second `if` should not be as indented as the first.